contrib/cmd/waitforjob/main.go

```diff
@@ -68,70 +79,83 @@ func (w *waitForJobOpts) Run() {
 
 	client, namespace, err := w.localClient()
 	if err != nil {
-		log.WithError(err).Fatal("failed to obtain client")
+		log.WithError(err).Fatal("Could not create a local client")
 	}
 
-	if err := w.waitForJobExistence(client, namespace); err != nil {
-		log.WithError(err).Fatal("job existence failed")
+	if w.jobType == uninstallJobType {
```

Comment:
My understanding is that, in the uninstall case, the wait-for may come too late and miss the uninstall job. That is why it passes when the uninstall job does not exist. The test should never miss the install job as that takes a lot longer to run. Perhaps passing on a missing uninstall job is not necessary any more as the uninstall job should take a significant chunk of time, too.

Reply:
So, if anything, neither the install case or the uninstall case should pass when the job does not exist, in my opinion.

Reply:
The reason we wait for the job to exist is that it won’t get created as soon as you perform the action like creating or deleting the cluster deployment. There’s the really small time between the action and the controller syncing the cluster deployment and creating the job. For create, the e2e has a 2 min sleep which seems wrong.

Reply:
Oh, I get your point now. The wait-for does not need to wait for the job to exist. The job can be created after the wait-for starts, and it will still be picked up.